src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml

```diff
@@ -922,8 +941,48 @@ end = struct
                 let value = compute lc in
                 res.(col_idx).(row_idx) <- value ;
                 Hashtbl.set internal_values ~key:var ~data:value ) ) ;
+
+    let map_runtime_tables = MapRuntimeTable.Table.create () in
+    let runtime_tables : Fp.t Kimchi_types.runtime_table array =
+      match sys.runtime_tables_cfg with
+      | Unfinalized_runtime_tables_cfg_rev _ ->
+          failwith
+            "Attempted to generate a witness for an unfinalized constraint \
+             system"
+      | Compiled_runtime_tables_cfg cfgs ->
+          Array.mapi cfgs ~f:(fun rt_idx { Kimchi_types.id; first_column } ->
+              let data =
+                Array.mapi first_column ~f:(fun i v ->
+                    ignore
+                    @@ MapRuntimeTable.Table.add map_runtime_tables ~key:(id, v)
+                         ~data:(i, rt_idx) ;
+                    (* default padding value for lookup *)
+                    Fp.zero )
+              in
+              let rt : Fp.t Kimchi_types.runtime_table = { id; data } in
+              rt )
+    in
+
+    (* Fill in the used entries of the runtime lookup tables. *)
+    List.iter sys.runtime_lookups ~f:(fun (id, (idx, v)) ->
+        let compute_value x = compute ([ (Fp.one, x) ], None) in
+        let vid = compute_value id in
+        let vidx = compute_value idx in
+        let vv = compute_value v in
+        (* FIXME: we should have a int32 here. We are not sure the ID will be a
+           int32. We should enforce that.
+           See https://github.com/MinaProtocol/mina/issues/13955
+        *)
+        let id_int32 = Int32.of_string @@ Fp.to_string vid in
+        (* FIXME: make a better exception. See
+           https://github.com/MinaProtocol/mina/issues/13954 *)
+        let i, rt_idx =
+          MapRuntimeTable.Table.find_exn map_runtime_tables (id_int32, vidx)
```

Comment:
This doesn't gracefully handle values from the fixed table

Reply:
See https://github.com/MinaProtocol/mina/pull/13939#discussion_r1309384554

Reply:
Mmh, let me have a second look.